src/libraries/System.Drawing.Common/src/System/Drawing/Imaging/ColorPalette.cs

```diff
@@ -61,14 +61,14 @@ internal void ConvertFromMemory(IntPtr memory)
 
             int size;
 
-            size = Marshal.ReadInt32((IntPtr)((long)memory + 4));  // Marshal.SizeOf(size.GetType())
+            size = Marshal.ReadInt32((nint)((long)memory + 4));  // Marshal.SizeOf(size.GetType())
```

Comment:
```suggestion
            size = *(int*)(pMemory + 4);
```
Also, add `byte* pMemory = (byte*)memory;`